### PR TITLE
feat: add `disable_style` option

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: #2984 
Ref: https://github.com/lfortran/lfortran/pull/3901#issuecomment-2066934356

### CODE REVIEW

File: tests/reference/disable_style-implicit10-d5777c8.stderr [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (gxyd) [inline tests/reference/disable_style-implicit10-d5777c8.stderr:1]: If we aren't raising any error, why is this file needed? Seems like some part even though these new lines character are a part of "errors" which I don't think should be there.

File: src/bin/lfortran.cpp [hunk 2]
```diff
@@ -2060,6 +2060,7 @@ int main_app(int argc, char *argv[]) {
     app.add_flag("--static", static_link, "Create a static executable");
     app.add_flag("--shared", shared_link, "Create a shared executable");
     app.add_flag("--no-warnings", compiler_options.no_warnings, "Turn off all warnings");
+    app.add_flag("--disable-style", compiler_options.disable_style, "Turn off style suggestions");
```
Comment (certik) [inline src/bin/lfortran.cpp:2063]: Isn't this a better option:
```suggestion
    app.add_flag("--no-style-warnings", compiler_options.disable_style, "Turn off style suggestions");
```
Later we can figure out some unified options, perhasp `-W`.

File: tests/reference/disable_style-implicit10-d5777c8.stderr (hunk 1 above)
Comment (certik) [reply tests/reference/disable_style-implicit10-d5777c8.stderr:1]: If the banner is disabled and style are disabled (but otherwise there are style diagnostic messages), it looks like we currently print the two empty lines to stderr. We should figure out where these empty lines are coming from and disable them. I think that can be done in a subsequent PR.

Comment (HarshitaKalani) [reply tests/reference/disable_style-implicit10-d5777c8.stderr:1]: Opened https://github.com/lfortran/lfortran/issues/3921 for this.

**Review by certik — APPROVED** [review]
I think this is good enough to merge. Thanks!
